### PR TITLE
feat: add mgconvco.ConvcoCheck

### PR DESCRIPTION
**Pull request by odsod**
Faster than commitlint and free of dependency on Node.JS.

### CODE REVIEW

**Review by Edholm — APPROVED** [review]
Nice! Works roughly the same?

**Comment by odsod** [comment]
@Edholm Yes, maybe a little less strict. But I'm looking into this one too now: https://github.com/cocogitto/cocogitto

**Comment by odsod** [comment]
It's more well-documented than convco, which makes me think it might outlive it. :smile_cat:

File: targets/mgconvco/targets.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const version = "0.3.7"
+
+var executable string
+
+func ConvcoCheck(ctx context.Context, rev string) error {
+	logger := mglog.Logger("convco-check")
+	ctx = logr.NewContext(ctx, logger)
+	mg.CtxDeps(ctx, prepare)
+	logger.Info("checking...", "rev", rev)
+	return sh.RunV(executable, "check", rev)
```
Comment (alethenorio) [inline targets/mgconvco/targets.go:27]: This is set on the prepare function but I don't see where it is called. How is the prepare function meant to be called?

**Review by alethenorio — APPROVED** [review]
Looks good. Haven't followed the latest changes so uncertain about how this is meant to work.

File: targets/mgconvco/targets.go (hunk 1 above)
Comment (alethenorio) [reply targets/mgconvco/targets.go:27]: More specifically I meant the "executable" variable

Comment (viktorvoltaire) [reply targets/mgconvco/targets.go:27]: It’s called via the `mg.CtxDeps`

**Review by viktorvoltaire — APPROVED** [review]
Awesome!

File: targets/mgconvco/targets.go (hunk 1 above)
Comment (alethenorio) [reply targets/mgconvco/targets.go:27]: Doh. Missed it somehow

**Comment by odsod** [comment]
@Edholm @alethenorio @viktorvoltaire I added cocogitto as an alterantive git commit checker - it works without any arguments (it reads commits from the latest git tag) and is a little more copy-pasteable (main/master doesn't matter).

**Review by Edholm — APPROVED** [review]
Nice!

**Review by viktorvoltaire — APPROVED** [review]
Awesome! 👍

**Comment by alethenorio** [comment]
@odsod how does it work for repos that don't have tags?

**Comment by odsod** [comment]
@alethenorio Then it works like without that flag - all commits from HEAD to the first commit

**Pull request merged**